core/chains/evm/evm_txm.go

```diff
@@ -0,0 +1,54 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/sqlx"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmconfig "github.com/smartcontractkit/chainlink/core/chains/evm/config"
+	httypes "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// GenericTxManager is type alias for txmgr.TxManager.
+// This is necessary because Golang doesn't allow embedding
+// txmgr.TxManager directly inside evmTxm struct.
+type GenericTxManager = txmgr.TxManager
```

Comment:
I am surprised this is necessary because an alias is supposed to literally be the same type. Are you sure there wasn't something else going on when you tried it?

Reply:
The error I see is on line 53.
If replace GenericTxManager by txmgr.TxManager, it complains of invalid field type.

Reply:
Ahh, I see how I can now avoid that error.
Instead of using txmgr.TxManager, I can directly use TxManager.

Works now. Fixing it. Sorry, didn't realize this is how golang work!

Reply:
@simsonraj : Related to your comment earlier. I just had to use TxManager directly, instead of txmgr.TxManager.

Reply:
@samsondav : On your question "Where are the comments on lines 18 thru 21?"

I removed it all. Was able to simplify usage as per above thread.